urld5c972ef81_noxx90_jflam_tgz-pJ8-fr_noxx90_jflam_BenchmarkJ8/src/fr/noxx90/jflam/renderer/impl/DefaultHistogramRenderer.java

```diff
@@ -0,0 +1,150 @@
+package fr.noxx90.jflam.renderer.impl;
+
+import java.awt.Color;
+import java.awt.geom.Point2D;
+import java.util.Iterator;
+
+import com.google.common.collect.Range;
+import com.google.common.collect.RangeMap;
+import com.google.common.collect.TreeRangeMap;
+
+import fr.noxx90.jflam.model.Flame;
+import fr.noxx90.jflam.model.Form;
+import fr.noxx90.jflam.model.Histogram;
+import fr.noxx90.jflam.model.impl.InMemoryHistogram;
+import fr.noxx90.jflam.renderer.HistogramRenderer;
+
+import static com.esotericsoftware.minlog.Log.*;
+
+public class DefaultHistogramRenderer extends HistogramRenderer
+{
+	protected RangeMap<Float, Form> rangeMap;
+	protected float weightSum;
+	protected float ratio;
+	protected float marginTop;
+	protected float marginLeft;
+	
+	@Override
+	public Histogram render(Flame flame, int width, int height, int quality, int sampling) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.